src/latch/executions.py

```diff
@@ -160,3 +160,60 @@ def report_nextflow_used_storage(used_bytes: int):
         """),
         {"argToken": token, "argUsedStorageBytes": used_bytes},
     )
+
+
+@dataclass
+class ExecutionMetadata:
+    execution_id: str
+    execution_name: str
+    workflow_id: str
+    workflow_name: str
+    workflow_version: str
+
+    @property
+    def url(self) -> str:
+        return f"https://console.latch.bio/workflows/{self.workflow_id}/executions/{self.execution_id}"
+
+
+def get_execution_metadata(token: Optional[str] = None) -> ExecutionMetadata:
+    if token is None:
+        token = os.environ.get("FLYTE_INTERNAL_EXECUTION_TOKEN")
+
+    if token is None or token == "":
+        raise RuntimeError(
+            "get_execution_metadata can only be called from within a running execution"
+        )
+
+    res = execute(
+        gql.gql(
+            """
+            query GetExecutionMetadata($token: String!) {
+                executionCreatorByToken(token: $token) {
```

Comment:
is this a creator? looks like an executionInfo?

Reply:
it is a creator, theres a join with `info`